### PR TITLE
Team name field

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description
Add a new field with the team name and allows to filter the result for team name.

## Reviewers 
- [ ] @jfavellar90 
- [ ] @diegomillan

### CODE REVIEW

File: edx_sga/sga.py [hunk 1]
```diff
@@ -902,13 +902,15 @@ def get_student_data():
                     'may_grade': instructor or not approved,
                     'annotated': force_text(state.get("annotated_filename", '')),
                     'comment': force_text(state.get("comment", '')),
-                    'finalized': is_finalized_submission(submission_data=submission)
+                    'finalized': is_finalized_submission(submission_data=submission),
+                    'team': self.get_team_name(student_module.student.username)
```
Comment (diegomillan) [inline edx_sga/sga.py]: Always leave a trailing `,` in every dict element. In the future if you modified the dict to add another element, then only one line of code will be affected therefore it is easier to review.

File: edx_sga/sga.py [hunk 2]
```diff
@@ -902,13 +902,15 @@ def get_student_data():
...
                 }
 
         return {
             'assignments': list(get_student_data()),
             'max_score': self.max_score(),
-            'display_name': force_text(self.display_name)
+            'display_name': force_text(self.display_name),
+            'teams_view': self.team_view
```
Comment (diegomillan) [inline edx_sga/sga.py]: Once again, trailing comma

File: edx_sga/sga.py [hunk 3]
```diff
@@ -1162,6 +1164,35 @@ def _user_image_url(self, user):
...
+        if not self.team_view:
+            return None
+
+        runtime = self.xmodule_runtime
+        user = runtime.service(self, 'user').get_current_user()
+        course_id = runtime.course_id
+
+        xblock_settings = self.get_xblock_settings()
+
+        try:
```
Comment (diegomillan) [inline edx_sga/sga.py]: Does it makes sense to extract this logic in another method??? then we could use it in any other method that needs to validate the same.

**Review by diegomillan — CHANGES_REQUESTED** [review]
Please check my comments @andrey-canon

File: edx_sga/sga.py (hunk 1 above)
Comment (andrey-canon) [reply edx_sga/sga.py]: check

File: edx_sga/sga.py (hunk 3 above)
Comment (andrey-canon) [reply edx_sga/sga.py]: Check

**Comment by andrey-canon** [comment]
@diegomillan  your recommendations have been applied in the last commit

**Review by diegomillan — APPROVED** [review]

**Comment by diegomillan** [comment]
Ready for merge @andrey-canon @jfavellar90

File: edx_sga/sga.py [hunk 4]
```diff
@@ -1162,6 +1152,38 @@ def _user_image_url(self, user):
         image_url = "{}{}".format(base_url, profile_image_url)
         return image_url
 
+    def get_team_name(self, username):
```
Comment (jfavellar90) [inline edx_sga/sga.py:1155]: Docstring

File: edx_sga/sga.py [hunk 5]
```diff
@@ -1162,6 +1152,38 @@ def _user_image_url(self, user):
...
+        course_id = runtime.course_id
+
+        api = self.api_rocket_chat()
+        team = api.get_user_team(course_id, username)
+        if team:
+            team = team[0]
+            return "-".join([team["topic_id"], team["name"]])
+        return None
+
+    def api_rocket_chat(self):
```
Comment (jfavellar90) [inline edx_sga/sga.py]: Docstring

File: edx_sga/sga.py [hunk 6]
```diff
@@ -1162,6 +1152,38 @@ def _user_image_url(self, user):
...
+
```
Comment (jfavellar90) [inline edx_sga/sga.py]: why the method is called api_rocket_chat but you return a teams API instance??

**Review by jfavellar90 — CHANGES_REQUESTED** [review]

**Comment by andrey-canon** [comment]
@jfavellar90  I have added the docstring and change the method name  :)

**Review by jfavellar90 — APPROVED** [review]

**Pull request merged**